pkg/projects/resource_project_user.go

```diff
@@ -0,0 +1,214 @@
+package projects
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/go-resty/resty/v2"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type User struct {
+	Name  string   `json:"name"`
+	Roles []string `json:"roles"`
+}
+
+type Comparable interface {
+	Hash() string
+	Equals(other Comparable) bool
+}
+
+func (user User) Hash() string {
+	return user.Name
+}
+
+func (a User) Equals(b User) bool {
+	return a.Hash() == b.Hash()
+}
+
+func contains(as []User, b User) bool {
+	log.Printf("[DEBUG] contains")
+	log.Printf("[TRACE] as: %+v\n", as)
+	log.Printf("[TRACE] b: %+v\n", b)
+
+	for _, a := range as {
+		log.Printf("[TRACE] a: %+v\n", a)
+		log.Printf("[TRACE] a.Equals(b): %+v\n", a.Equals(b))
+		if a.Equals(b) {
+			return true
+		}
+	}
+	return false
+}
+
+// Not the most efficient way to determine the slices intersection but this suffices for the small-ish number of items
+var usersIntersection = func(as []User, bs []User) []User {
+	log.Printf("[DEBUG] usersIntersection")
+	log.Printf("[TRACE] as: %+v\n", as)
+	log.Printf("[TRACE] bs: %+v\n", bs)
+
+	intersection := make([]User, 0)
+
+	for _, a := range as {
+		if contains(bs, a) {
+			intersection = append(intersection, a)
+		}
+	}
+
+	return intersection
+}
+
+// Not the most efficient way to determine the slices differences but this suffices for the small-ish number of items
+var usersDifference = func(as []User, bs []User) []User {
+	difference := make([]User, 0)
+
+	for _, a := range as {
+		if !contains(bs, a) {
+			difference = append(difference, a)
+		}
+	}
+
+	return difference
+}
+
+type Users struct {
+	Members []User
+}
+
+const projectUsersUrl = "/access/api/v1/projects/%s/users/"
+
+var unpackUsers = func(data *schema.ResourceData) (string, Users, error) {
+	d := &ResourceData{data}
+	projectKey := d.getString("key", false)
+
+	users := Users{
+		Members: make([]User, 0),
```

Comment:
So, this will give you a pointer but it's not necessary. By default you would have an empty slice if not set. In fact, if you make the next bit of code a function, you can do `Members: getMembers(d)`